src/Providers/CorrelationIdServiceProvider.php

```diff
@@ -32,4 +40,25 @@ function ($service, $app) {
             }
         );
     }
+
+    /**
+     * @param Kernel $kernel
+     * @return void
+     */
+    private function pushGlobalMiddleware(Kernel $kernel): void
+    {
+        if (!config('correlation_id.include_route_middleware')) {
```

Comment:
Maybe we could extract this to a separate method, we would then have something like this:
```suggestion
        if (!$this->shouldIncludeRouteMiddleware()) {
```

Reply:
Done.